### PR TITLE
Refactor e2e dashboard helpers to TypeScript

### PR DESCRIPTION
**Pull request by kamilmielnik**
### Description

- Refactors the following files to TypeScript:
  - `e2e/support/commands/api/dashboard.js`
  - `e2e/support/helpers/e2e-dashboard-helpers.js`
- Extracts `archiveDashboard` & `createDashboard` helpers and deprecates `cy.archiveDashboard` & `cy.createDashboard` commands

### How to verify

CI is green.

### CODE REVIEW

File: frontend/src/metabase-types/api/dashboard.ts [hunk 1]
```diff
@@ -252,6 +252,7 @@ export type CreateDashboardRequest = {
   cache_ttl?: number;
   collection_id?: CollectionId | null;
   collection_position?: number | null;
+  tabs?: DashboardTab[];
```
Comment (kamilmielnik) [inline frontend/src/metabase-types/api/dashboard.ts:255]: We were using this attribute in tests

File: e2e/support/helpers/e2e-dashboard-helpers.ts [hunk 2]
```diff
@@ -19,14 +101,10 @@ export function getDashboardCard(index = 0) {
   return getDashboardCards().eq(index);
 }
 
-export function ensureDashboardCardHasText(text, index = 0) {
+export function ensureDashboardCardHasText(text: string, index = 0) {
   cy.findAllByTestId("dashcard").eq(index).should("contain", text);
 }
 
-function getDashboardApiUrl(dashId) {
```
Comment (kamilmielnik) [inline e2e/support/helpers/e2e-dashboard-helpers.ts:26]: Inlined, used only in 2 places

File: e2e/support/helpers/e2e-dashboard-helpers.ts [hunk 3]
```diff
@@ -1,13 +1,95 @@
...
+   */
+  wrapId?: boolean;
+  /**
+   * Alias a dashboard id in order to use it later with `cy.get("@" + alias).
+   * Defaults to "dashboardId".
+   */
+  idAlias?: string;
+}
+
+export const createDashboard = (
```
Comment (kamilmielnik) [inline e2e/support/helpers/e2e-dashboard-helpers.ts:35]: Extracted from `e2e/support/commands/api/dashboard.js`

File: e2e/support/helpers/e2e-dashboard-helpers.ts [hunk 4]
```diff
@@ -1,13 +1,95 @@
...
+          auto_apply_filters,
+          enable_embedding,
+          embedding_params,
+          dashcards,
+        });
+      }
+    });
+};
+
+export const archiveDashboard = (id: DashboardId) => {
```
Comment (kamilmielnik) [inline e2e/support/helpers/e2e-dashboard-helpers.ts:73]: Extracted from `e2e/support/commands/api/dashboard.js`

File: e2e/support/helpers/e2e-dashboard-helpers.ts [hunk 5]
```diff
@@ -1,13 +1,95 @@
...
+    embedding_params,
+    dashcards,
+    ...restDashboardDetails
+  } = dashboardDetails;
+  const { wrapId = false, idAlias = "dashboardId" } = options;
+
+  cy.log(`Create a dashboard: ${name}`);
+
+  // For all the possible keys, refer to `src/metabase/api/dashboard.clj`
+  return cy
```
Comment (kamilmielnik) [inline e2e/support/helpers/e2e-dashboard-helpers.ts:52]: Added a `return` here

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js [hunk 6]
```diff
@@ -2358,16 +2359,6 @@ describe("issue 31662", () => {
   });
 });
 describe("issue 38245", () => {
-  function createDashboardWithTabs({ dashcards, tabs, ...dashboardDetails }) {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js:2361]: Duplicated definition, existing helper function was identical

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**